src/napari_matplotlib/scatter.py

```diff
@@ -214,8 +198,9 @@ def _get_data(self) -> Tuple[List[np.ndarray], str, str]:
         return data, x_axis_name, y_axis_name
 
     def _on_update_layers(self) -> None:
-        """This is called when the layer selection changes
-        by self.update_layers().
+        """
+        This is called when the layer selection changes by
+        ``self.update_layers()``.
         """
         if self._key_selection_widget is not None:
```

Comment:
```suggestion
        if hasattr(self, "_key_selection_widget") and self._key_selection_widget is not None:
```

I think this would fix the tests, which are failing because `self.update_layers()`, which needs to be called in the `ScatterBase` init. However, in the future, I think it would be best to refactor a bit such that the plotting code is in the base class and the layer selection code is in the subclass. This will give more flexibility in how layer selection is performed (e.g., layer can be set from a combobox and would be independent of the layer that is selected in the layer list)